Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/DirectoryInitialization.java

```diff
@@ -48,31 +45,28 @@
     CANT_FIND_EXTERNAL_STORAGE
   }
 
-  public DirectoryInitializationService()
+  public static void start(Context context)
   {
-    // Superclass constructor is called to name the thread on which this service executes.
-    super("DirectoryInitializationService");
+    // Can take a few seconds to run, so don't block UI thread.
+    //noinspection TrivialFunctionalExpressionUsage
+    ((Runnable) () -> init(context)).run();
   }
 
-  public static void startService(Context context)
+  private static void init(Context context)
   {
-    Intent intent = new Intent(context, DirectoryInitializationService.class);
-    context.startService(intent);
-  }
+    if (isDolphinDirectoryInitializationRunning.get())
```

Comment:
You should use `compareAndSet` instead of `get` followed by `set` in order to avoid race conditions.

Reply:
Fixed